recipes/armadillo/all/conanfile.py

```diff
@@ -91,6 +91,14 @@ def config_options(self):
             self.options.use_blas = "framework_accelerate"
             self.options.use_lapack = "framework_accelerate"
 
+        # According with the CMakeLists file in armadillo, MinGW doesn't correctly handle thread_local.
+        # If any of MINGW, MSYS, CYGWIN or MSVC are True in during cmake configure, the ARMA_USE_EXTERN_RNG option will be set to false.
+        # Therefore, in these cases we remove the `use_extern_rng` option in conan
+        from conans.client.tools.win import CYGWIN, MSYS2, MSYS
+        windows_subsystem = tools.os_info.detect_windows_subsystem()
+        if windows_subsystem in [MSYS, MSYS2, CYGWIN] or self.settings.compiler == "Visual Studio":
+            del self.options.use_extern_rng
```

Comment:
honestly, just remove this option in the condition `if self.settings.os == "Windows"` above.

Reply:
This was the thought I had [above](https://github.com/conan-io/conan-center-index/pull/7919#issuecomment-999229047). My uncertainty was how WSL would be treated in a case like this. If that's not an issue, I'm inclined to agree to simplify this to a check for Windows.

Reply:
Let's go simple, if someone runs into issues they'll probably know more and we can always revisit this